src/renderer/utils/pst-viewer.ts

```diff
@@ -0,0 +1,94 @@
+import type { PstContent } from "@common/modules/pst-extractor/type";
+import type {
+    CirclePackingSvgProps,
+    ComputedDatum,
+} from "@nivo/circle-packing";
+
+import type {
+    DefaultViewerObject,
+    MailViewerObject,
+    ViewerObjectChild,
+} from "./pst-extractor";
+import {
+    isMailViewerObject,
+    isToDeleteFolder,
+    isToKeepFolder,
+} from "./pst-extractor";
+
+export const getChildrenToDeleteIds = (
+    children: ViewerObjectChild[],
+    markedToKeepIds: string[]
+): string[] =>
+    children
+        .filter(
+            (child: ViewerObjectChild) =>
+                !isToKeepFolder(child.id, markedToKeepIds)
+        )
+        .map((child) => child.id);
+
+export const getChildrenToKeepIds = (
+    children: ViewerObjectChild[],
+    markedToDeleteIds: string[]
+): string[] =>
+    children
+        .filter(
+            (child: ViewerObjectChild) =>
+                !isToDeleteFolder(child.id, markedToDeleteIds)
+        )
+        .map((child) => child.id);
+
+export const getUntagChildrenIds = (
+    children: ViewerObjectChild[],
+    markedToDeleteIds: string[],
+    markedToKeepIds: string[]
+): string[] =>
+    children
+        .filter(
+            (child: ViewerObjectChild) =>
+                !isToDeleteFolder(child.id, markedToDeleteIds) &&
+                !isToKeepFolder(child.id, markedToKeepIds)
+        )
+        .map((child) => child.id);
+
+export const getColorFromTrimester = (
+    node: Omit<ComputedDatum<MailViewerObject<string>>, "color" | "fill">
+): number => {
+    const month = node.data.email.receivedDate?.getMonth() ?? 1; // January
+
+    return month <= 3 ? 0.45 : month <= 6 ? 0.65 : month <= 9 ? 0.85 : 1; // TODO: magic number
+};
+
+////////////
+// Redeclare because ResponsiveCirclePackingProps is not exported.
+type ResponsiveCirclePackingProps<TRawDatum> = Partial<
+    Omit<CirclePackingSvgProps<TRawDatum>, "data" | "height" | "width">
+> &
+    Pick<CirclePackingSvgProps<TRawDatum>, "data">;
+
+export type CirclePackingCommonProps = Partial<
+    ResponsiveCirclePackingProps<DefaultViewerObject<string>>
+> & {
+    id: keyof PstContent;
+    value: keyof PstContent;
+};
+
+export const commonProperties: CirclePackingCommonProps = {
+    enableLabels: true,
+    id: "id",
+    isInteractive: true,
+    label: (node) =>
+        isMailViewerObject(node.data)
+            ? `${node.data.email.receivedDate?.getDay()}-${node.data.email.receivedDate?.getMonth()}-${node.data.email.receivedDate?.getFullYear()}`
+            : node.data.name,
+    labelsFilter: (label) => label.node.height === 0,
+    labelsSkipRadius: 16,
+    motionConfig: "slow",
+    padding: 2,
+    value: "size",
+};
+
+export const sanitizeMailDate = (date: Date): string =>
```

Comment:
📅 use [Date.prototype.toLocalDateString()](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) or event better, [Intl.DateTimeFormat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat) in combinaison with i18n module

Reply:
Yes, this sanitize function is a bit awkward and "under-engineer"... 😅 but yes, should be modify with i18n module when translation will be under construction!